pkg/controllers/provisioningv2/cluster/controller.go

```diff
@@ -225,9 +224,12 @@ func (h *handler) createCluster(cluster *v1.Cluster, status v1.ClusterStatus, sp
 	return h.createNewCluster(cluster, status, spec)
 }
 
-func mgmtClusterName(clusterNamespace, clusterName string) string {
-	hash := sha256.Sum256([]byte(clusterNamespace + "/" + clusterName))
-	return name.SafeConcatName("c", "m", hex.EncodeToString(hash[:])[:8])
+func mgmtClusterName() (string, error) {
```

Comment:
This is worth to have @ibuildthecloud look at it. I think it has a reason to use hash to determine the name, and there might be code elsewhere that use `clusterNamespace` and `clusterName` to look for v3.Cluster from v1.cluster. Using the random name might break that use case. If not then this change might be fine.

Reply:
@StrongMonkey I did speak to him about this, but yes, I think it is worth him looking at it.

Reply:
It should be fine.  The only that that matters is that multiple runs of the controller produce the same name so that apply doesn't create/delete the cluster object.  All other code should reference status.ClusterName for the name relationship between prov->mgmt clusters.